go/controller/post_token_test.go

```diff
@@ -0,0 +1,168 @@
+	"go.uber.org/mock/gomock"
+)
+
+func TestPostToken(t *testing.T) {
+	t.Parallel()
+
+	userID := uuid.MustParse("db477732-48fa-4289-b694-2886a646b6eb")
+	token := uuid.MustParse("1fb17604-86c7-444e-b337-09a644465f2d")
+	hashedToken := `\x9698157153010b858587119503cbeef0cf288f11775e51cdb6bfd65e930d9310`
+
+	cases := []testRequest[api.PostTokenRequestObject, api.PostTokenResponseObject]{
+		{
+			name:   "simple",
+			config: getConfig,
+			db: func(ctrl *gomock.Controller) controller.DBClient {
+				mock := mock.NewMockDBClient(ctrl)
+
+				mock.EXPECT().GetUserByRefreshTokenHash(
+					gomock.Any(),
+					sql.GetUserByRefreshTokenHashParams{
+						RefreshTokenHash: sql.Text(hashedToken),
+						Type:             "regular",
+					},
+				).Return(getSigninUser(userID), nil)
+
+				mock.EXPECT().RefreshTokenAndGetUserRoles(
+					gomock.Any(),
+					cmpDBParams(sql.RefreshTokenAndGetUserRolesParams{
+						RefreshTokenHash: sql.Text(hashedToken),
+						ExpiresAt: sql.TimestampTz(
+							time.Now().Add(time.Duration(2592000) * time.Second),
+						),
+					}),
+				).Return([]string{"user", "me"}, nil)
+
+				return mock
+			},
+			request: api.PostTokenRequestObject{
+				Body: &api.RefreshTokenRequest{
+					RefreshToken: token.String(),
+				},
+			},
+			expectedResponse: api.PostToken200JSONResponse{
+				Session: &api.Session{
+					AccessToken:          "",
+					AccessTokenExpiresIn: time.Now().Add(900 * time.Second).Unix(),
+					RefreshToken:         "1fb17604-86c7-444e-b337-09a644465f2d",
+					User: &api.User{
+						AvatarUrl:           "",
+						CreatedAt:           time.Now(),
+						DefaultRole:         "user",
+						DisplayName:         "Jane Doe",
+						Email:               "jane@acme.com",
+						EmailVerified:       true,
+						Id:                  "db477732-48fa-4289-b694-2886a646b6eb",
+						IsAnonymous:         false,
+						Locale:              "en",
+						Metadata:            map[string]any{},
+						PhoneNumber:         "",
+						PhoneNumberVerified: false,
+						Roles:               []string{"user", "me"},
+					},
+				},
+			},
+			expectedJWT: &jwt.Token{
+				Raw:    "",
+				Method: jwt.SigningMethodHS256,
+				Header: map[string]any{
+					"alg": "HS256",
+					"typ": "JWT",
+				},
+				Claims: jwt.MapClaims{
+					"exp": float64(time.Now().Add(900 * time.Second).Unix()),
+					"https://hasura.io/jwt/claims": map[string]any{
+						"x-hasura-allowed-roles":     []any{"user", "me"},
+						"x-hasura-default-role":      "user",
+						"x-hasura-user-id":           "db477732-48fa-4289-b694-2886a646b6eb",
+						"x-hasura-user-is-anonymous": "false",
+					},
+					"iat": float64(time.Now().Unix()),
+					"iss": "hasura-auth",
+					"sub": "db477732-48fa-4289-b694-2886a646b6eb",
+				},
+				Signature: []byte{},
+				Valid:     true,
+			},
+			customClaimer: nil,
+			emailer:       nil,
+			hibp:          nil,
+			jwtTokenFn:    nil,
+		},
+		{
+			name: "invalid refresh token",
```

Comment:
I'd also suggest testing cases like:

1. user is anonymous
2. user is disabled


Mostly because you are reusing `GetUserByRefreshTokenHash` which checks for those and may fail in case you don't want to fail (i.e. it should work for anonymous users but fail for disabled).